packages/shopify/src/types/metafields.ts

```diff
@@ -0,0 +1,39 @@
+type LiteralUnion<T extends string> = T | Omit<T, T>
+
+export type MetafieldTypes =
+  | 'integer'
+  | 'boolean'
+  | 'color'
+  | 'json'
+  | 'date'
+  | 'file_reference'
+  | 'date_time'
+  | 'dimension'
+  | 'multi_line_text_field'
+  | 'number_decimal'
+  | 'number_integer'
+  | 'page_reference'
+  | 'product_reference'
+  | 'rating'
+  | 'single_line_text_field'
+  | 'url'
+  | 'variant_reference'
+  | 'volume'
+  | 'weight'
+  | 'list.color'
+  | 'list.date'
+  | 'list.date_time'
+  | 'list.dimension'
+  | 'list.file_reference'
+  | 'list.number_integer'
+  | 'list.number_decimal'
+  | 'list.page_reference'
+  | 'list.product_reference'
+  | 'list.rating'
+  | 'list.single_line_text_field'
+  | 'list.url'
+  | 'list.variant_reference'
+  | 'list.volume'
+  | 'list.weight'
+
+export type MetafieldType = LiteralUnion<MetafieldTypes>
```

Comment:
What does this type do? I tried it out locally and using this type accepts any string.

Reply:
Yes, since extends `string` and we use utils like `.split('.')` later on. It was not used to constraint specific "types", it was more used for autocomplete in the switch.